airflow-core/src/airflow/migrations/versions/0047_3_0_0_add_dag_versioning.py

```diff
@@ -84,165 +63,15 @@ def upgrade():
-        AND dag_id in (SELECT dag_id FROM dag)
-    """
-    rows = _get_rows(stmt, conn)
-    # Insert the missing rows from serialized_dag to dag_code
-    stmt = sa.text("""
-        INSERT INTO dag_code (dag_version_id, dag_id, fileloc, fileloc_hash, source_code, last_updated, created_at)
-        VALUES (:dag_version_id, :dag_id, :fileloc, :fileloc_hash, :source_code, :last_updated, :created_at)
-    """)
-    for row in rows:
-        try:
-            source_code = DagCode.get_code_from_file(row.fileloc)
-        except FileNotFoundError:
-            source_code = "source_code"
-        conn.execute(
-            stmt.bindparams(
-                dag_version_id=row.dag_version_id,
-                dag_id=row.dag_id,
-                fileloc=row.fileloc,
-                fileloc_hash=row.fileloc_hash,
-                source_code=source_code,
-                last_updated=timezone.utcnow(),
-                created_at=timezone.utcnow(),
-            )
-        )
-
-    stmt = "SELECT dag_id, fileloc FROM dag_code"
-    rows = _get_rows(stmt, conn)
-    stmt = sa.text("""
-                    UPDATE dag_code
-                    SET id = :_id,
-                        dag_id = :dag_id,
-                        source_code = :source_code,
-                        source_code_hash = :source_code_hash
-                    WHERE dag_id = :dag_id AND id IS NULL
-                """)
-    for row in rows:
-        id = uuid7()
-        if conn.dialect.name != "postgresql":
-            id = id.hex
-        else:
-            id = str(id)
-        try:
-            source_code = DagCode.get_code_from_file(row.fileloc)
-        except FileNotFoundError:
-            source_code = "source_code"
-        conn.execute(
-            stmt.bindparams(
-                _id=id,
-                source_code_hash=DagCode.dag_source_hash(source_code),
-                source_code=source_code,
-                dag_id=row.dag_id,
-            )
-        )
-
     with op.batch_alter_table("dag_code") as batch_op:
-        batch_op.alter_column("dag_id", existing_type=StringID(), nullable=False)
-        batch_op.alter_column("id", existing_type=UUIDType(binary=False), nullable=False)
+        batch_op.drop_constraint("dag_code_pkey", type_="primary")
+        batch_op.drop_column("fileloc_hash")
+        batch_op.add_column(sa.Column("id", UUIDType(binary=False), nullable=False))
         batch_op.create_primary_key("dag_code_pkey", ["id"])
-        batch_op.alter_column("dag_version_id", existing_type=UUIDType(binary=False), nullable=False)
+        batch_op.add_column(sa.Column("dag_version_id", UUIDType(binary=False), nullable=False))
+        batch_op.add_column(sa.Column("source_code_hash", sa.String(length=32), nullable=False))
+        batch_op.add_column(sa.Column("dag_id", StringID(), nullable=False))
+        batch_op.add_column(sa.Column("created_at", TIMESTAMP(), default=timezone.utcnow, nullable=False))
```

Comment:
Why can we change the `alter_column` calls to `add_column`?

Reply:
Thank you for reviewing.

So what's going on here is, previously we did things in two steps --- add column and then update data, then alter column (to set it to be not nullable)

But now I collapse it into one operation since we first truncate the tables (so there is no data to migrate)